### ISSUE TITLE
[Feature request] Disable dependencies

### ISSUE THREAD

**Issue by vRallev** [KeepSafe/dexcount-gradle-plugin#270]
This could be especially helpful for library modules. I'd like to know how many methods a module has. In other words: how many methods does this module alone contribute to my final apk excluding its dependencies?

It would be great to have a flag for this.

**Comment by benjamin-bader** [comment]
This would be a nice feature to have!

The problem is that the input to this plugin is .dex files, from which we can obtain no module information at all.  We could perhaps introspect the Gradle task graph and discover all the modules being built, then assemble mappings from class/method -> module, then provide that as input to the method-counting task.  It'd be a fair bit of work, but in principle it seems like something that could work.

I have basically no time for OSS work right now, and so I can't commit to doing this.  Would be happy to take a PR!

Note:
You could of course make the module in question produce an AAR, which we know how to handle, and you'd get a method count for just that code.  I don't recommend this because in general counting methods of AARs is unreliable; it does not reflect the true cost in methods to the final product (unless of course the module has no dependencies and the consuming APK uses every single one of the module's methods).  Even in the case of a pure Java module, the counts you'd obtain are inaccurate because many of the method references would be to system-provided methods that are already referenced by consuming modules, and so wouldn't count a second time.

TL;DR: if you decide to make your module an AAR and only apply the plugin there, the counts you will receive are bogus - they omit the costs of dependencies, and overstate the cost of the code in the module itself.

**Comment by vRallev** [comment]
After posting the issue I dug in a little more myself and missed a couple of things you've explained well. I was only thinking of declared methods and fields and not referenced methods. I misunderstood how the dex reference method count works.

That changes the scope completely of this ticket and would be simple to implement with javap and the `classes.jar` file. It would even work for java modules. 

I've prototyped this, but it's quite slow to call `javap` for each class from the jar file. It probably would be faster to only scan the source files, but also less accurate. Is this something you'd be interested in nonetheless?

**Comment by benjamin-bader** [comment]
It occurs to me that, assuming you have distinct packages for your modules, the info you seek is already in the plugin's output file.

Your idea is doable, and probably easier to accomplish with e.g. javassist or bytebuddy or some other bytecode library.  If you'd like to take this on, I'd accept it in the same spirit that we have .aar counting, with the caveat that the output isn't directly relatable to the final method count.

**Comment by vRallev** [comment]
@benjamin-bader I've created #271.
